src/libs/sfdk/buildengine.cpp

```diff
@@ -319,8 +319,10 @@ void BuildEngine::importPrivateGpgKey(const QString &id,
         const Functor<bool, QString> &functor)
 {
     QString errorString;
-    QTC_ASSERT(isGpgAvailable(&errorString),
-            QTimer::singleShot(0, context, std::bind(functor, false, errorString)); return);
+    QTC_ASSERT(isGpgAvailable(&errorString), {
+        BatchComposer::enqueueCheckPoint(context, std::bind(functor, false, errorString));
```

Comment:
"Fix async returns from asserts" - not really the best possible commit message. Why where they broken? I would appreciate "Sfdk: use enqueueCheckPoint() instead of QTimer::singleShot" and then on the next line "QTimer::singleShot no longer works because of x" more.

Reply:
They never worked. I realized it some time ago already but only now there was an opportunity to fix it. They fulfilled the requirement of being asynchronous, but failed at fulfilling the requirement of being ordered.

I will add that info to the commit message but I will leave the summary as is to describe the "what" rather than the "how" and fit into 50 chars limit more easily :)